crates/pop-contracts/src/utils/metadata.rs

```diff
@@ -42,14 +42,15 @@ pub enum FunctionType {
 /// Extracts a list of smart contract messages parsing the metadata file.
 ///
 /// # Arguments
-/// * `path` -  Location path of the project.
+/// * `path` -  Location path of the project or contract metadata file.
 pub fn get_messages(path: &Path) -> Result<Vec<ContractFunction>, Error> {
-	let cargo_toml_path = match path.ends_with("Cargo.toml") {
-		true => path.to_path_buf(),
-		false => path.join("Cargo.toml"),
+	let contract_artifacts = if path.is_dir() || path.ends_with("Cargo.toml") {
+		let cargo_toml_path =
+			if path.ends_with("Cargo.toml") { path.to_path_buf() } else { path.join("Cargo.toml") };
+		ContractArtifacts::from_manifest_or_file(Some(&cargo_toml_path), None)?
+	} else {
+		ContractArtifacts::from_manifest_or_file(None, Some(&path.to_path_buf()))?
```

Comment:
If I use a contract metadata location in the argument `--path`, `from_manifest_or_file` seem to still try to locate the manifest.

Reply:
Good catch! Fixed here https://github.com/r0gue-io/pop-cli/pull/347/commits/53f350505f6d33eadf93f505c4640b4c5350b509 and added some tests